lib/stdlib/src/gen_server.erl

```diff
@@ -2264,6 +2283,15 @@ loop(Parent, Name, State, CbCache, infinity, HibernateAfterTimeout, Debug) ->
             loop(Parent, Name, State, CbCache, hibernate, HibernateAfterTimeout, Debug)
     end;
 
+loop(Parent, Name, State, CbCache, {timeout, Time, TimeoutMsg} = Timeout, HibernateAfterTimeout, Debug) ->
```

Comment:
Perhaps `Time =:= infinity` should be handled here instead of in `handle_msg/7`, `handle_common_reply/8` below, and two places above; `init_it/5` and `enter_loop/5` (missing).

Reply:
TBH, I had it this way before, then thought having it in the `handle_msg` and friends functions would be better. But you're right, especially when combined with what @michalmuskala suggested. I'll change it back later.